src/modules/filesystem.md

```diff
@@ -21,6 +21,15 @@ The `crate` root is in:
 * `src/lib.rs` (for a library crate)
 * `src/main.rs` (for a binary crate)
 
+Modules defined in files can be documented, too, using "inner doc comments".
+These document the item that contains them -- in this case, a module.
+
+```rust,editable,compile_fail
+//! This module implements the garden, including a highly performant germination
+//! implementation.
+pub mod vegetables;
```

Comment:
I'm not sure why the submodule here is necessary, but if you want to have it then I suggest at least adding a blank line before it, to make clear that the Rustdoc comment is not associated with it.